packages/eslint-plugin/src/rules/member-ordering.ts

```diff
@@ -40,6 +86,215 @@ const allMemberTypes = ['field', 'method', 'constructor'].reduce<string[]>(
   [],
 );
 
+const neverConfig = {
+  type: 'string',
+  enum: ['never'],
+};
+
+const allMemberTypesArrayConfig = {
+  type: 'array',
+  items: {
+    enum: allMemberTypes,
+  },
+};
+
+const allMemberTypesDefaultConfig = {
+  type: 'string',
+  enum: ['never'],
+};
+
+const allMemberTypesObjectConfig = {
+  type: 'object',
+  properties: {
+    memberTypes: {
+      oneOf: [allMemberTypesDefaultConfig, allMemberTypesArrayConfig],
+    },
+    order: {
+      type: 'string',
+      enum: ['alphabetically'],
+    },
+  },
+  additionalProperties: false,
+};
+
+const functionExpressions = [
+  AST_NODE_TYPES.FunctionExpression,
+  AST_NODE_TYPES.ArrowFunctionExpression,
+];
+
+/**
+ * Gets the node type.
+ * @param node the node to be evaluated.
+ */
+function getNodeType(
+  node: TSESTree.ClassElement | TSESTree.TypeElement,
+): string | null {
+  switch (node.type) {
+    case AST_NODE_TYPES.MethodDefinition:
+      return node.kind;
+    case AST_NODE_TYPES.TSMethodSignature:
+      return 'method';
+    case AST_NODE_TYPES.TSConstructSignatureDeclaration:
+      return 'constructor';
+    case AST_NODE_TYPES.ClassProperty:
+      return node.value && functionExpressions.includes(node.value.type)
+        ? 'method'
+        : 'field';
+    case AST_NODE_TYPES.TSPropertySignature:
+      return 'field';
+    default:
+      return null;
+  }
+}
+
+/**
+ * Gets the member name based on the member type.
+ *
+ * @param node the node to be evaluated.
+ * @param sourceCode
+ */
+function getMemberName(
```

Comment:
am I mistaken - isn't this essentially the same thing as `getIdentifier`?

Reply:
Almost. The difference is the handling of the `AST_NODE_TYPES.MethodDefinition` differed but it looks as if this is not really needed. I've removed `getIdentifier`.